build.psm1

```diff
@@ -489,9 +489,15 @@ Fix steps:
     }
 
     # handle TypeGen
-    if ($TypeGen -or -not (Test-Path "$PSScriptRoot/src/System.Management.Automation/CoreCLR/CorePsTypeCatalog.cs")) {
+    $incFileName = 'powershell.inc'
+    if (-not [string]::IsNullOrEmpty($Runtime)) {
+        # File name must be different for Windows and Linux to allow build on Windows and WSL.
+        # Append the first 3 characters fo the Runtime only because otherwise dotnet will crash due to the separation characters. The underscore is only for better readability.
+        $incFileName = "$($incFileName)_$($Runtime.Substring(0,3)).inc"
+    }
+    if ($TypeGen -or -not (Test-Path "$PSScriptRoot/TypeCatalogGen/$($incFileName)")) {
```

Comment:
`$($incFileName)` can be replaced by `$incFileName`

Reply:
No, this is one of the gotchas about PowerShell string interpolation because PowerShell allows underscores for variable names. Therefore PowerShell would then look for a variable named `incFileName_` instead of `incFileName`. As a general rule, I always wrap variables in a string inside `$()` because it also safeguards against cases like calling a property on an object...

Reply:
In the case of `"$($incFileName)_xxx.inc"`, you can replace it with `"${incFileName}_xxx.inc"`